extensions/pkg/controller/extension/actuator.go

```diff
@@ -12,16 +12,72 @@ import (
 	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
 )
 
-// Actuator acts upon Extension resources.
+// Actuator acts upon [extensionsv1alpha1.Extension] resources.
+//
+// Actuator implementations plug into the generic reconciliation flow for
+// functionality that does not belong to the specialized resources such as
+// Bastion, ControlPlane, Network, etc.
 type Actuator interface {
-	// Reconcile the Extension resource.
+	// Reconcile reconciles the [extensionsv1alpha1.Extension] resource.
+	//
+	// Implementations of the [Actuator] interface should take care for
+	// creating or updating any resources (e.g. Managed Resources, Secrets,
+	// etc.), which the extension manages, so that they are brought to their
+	// desired state.
+	//
+	// This method may be invoked during the following conditions:
+	//
+	// - Initial creation of the [extensionsv1alpha1.Extension] resource
+	// - Updates to the extension's spec
+	// - Periodic reconciliation as configured by the [AddArgs.Resync] duration
```

Comment:
This is not explicitly explains for other actuators. I would either leave it out because it's common knowledge, or add similar statements to the other interfaces.